File: shared/react-native/android/app/src/main/java/io/keybase/ossifrage/util/DNSNSFetcher.java
package io.keybase.ossifrage.util;

import java.lang.reflect.Method;
import java.util.ArrayList;

public class DNSNSFetcher implements go.keybase.ExternalDNSNSFetcher {
    public byte[] getServers() {
        try {
            Class<?> SystemProperties = Class.forName("android.os.SystemProperties");
            Method method = SystemProperties.getMethod("get", new Class[] { String.class });
            ArrayList<String> servers = new ArrayList<String>();
            for (String name : new String[] { "net.dns1", "net.dns2", "net.dns3", "net.dns4", }) {
                String value = (String) method.invoke(null, name);
                if (value != null && !"".equals(value) && !servers.contains(value))
                    servers.add(value);
            }
            String srvStr = new String();
            for (int i = 0; i < servers.size(); i++) {
                srvStr += servers.get(i);
                if (i < servers.size()-1) {
                    srvStr+=",";
                }
            }
            return srvStr.getBytes();
        } catch (Exception e) {
            return "".getBytes();
        }
    }
}

File: shared/android/app/src/main/java/io/keybase/ossifrage/util/DNSNSFetcher.kt
package io.keybase.ossifrage.util

import keybase.ExternalDNSNSFetcher

class DNSNSFetcher : ExternalDNSNSFetcher {
    override fun getServers(): ByteArray {
        return try {
            val SystemProperties = Class.forName("android.os.SystemProperties")
            val method = SystemProperties.getMethod("get", *arrayOf<Class<*>>(String::class.java))
            val servers = ArrayList<String>()
            for (name in arrayOf("net.dns1", "net.dns2", "net.dns3", "net.dns4")) {
                val value = method.invoke(null, name) as String?
                if (value != null && "" != value && !servers.contains(value)) servers.add(value)
            }
            val srvStr = servers.joinToString(",")
            srvStr.toByteArray(Charsets.UTF_8)
        } catch (e: Exception) {
            "".toByteArray()
        }
    }
}
